Some tests fail with Swift 5.0 on macOS

```terminal.sh-session
$ xcrun --toolchain org.swift.5020181225a swift test
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Sources/Yams/Parser.swift:139:14: warning: 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
        data.withUnsafeBytes { bytes in
             ^
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:139:28: warning: 'init(bytes:)' is deprecated: use `init(_:)` instead
        _testRoundTrip(of: Data(bytes: [0xDE, 0xAD, 0xBE, 0xEF]), expectedYAML: "3q2+7w==\n")
                           ^
[3/3] Linking ./.build/x86_64-apple-macosx/debug/YamsPackageTests.xctest/Contents/MacOS/YamsPackageTests
Test Suite 'All tests' started at 2018-12-26 19:32:12.867
Test Suite 'YamsPackageTests.xctest' started at 2018-12-26 19:32:12.867
Test Suite 'ConstructorTests' started at 2018-12-26 19:32:12.867
Test Case '-[YamsTests.ConstructorTests testBinary]' started.
Test Case '-[YamsTests.ConstructorTests testBinary]' passed (0.240 seconds).
Test Case '-[YamsTests.ConstructorTests testBool]' started.
Test Case '-[YamsTests.ConstructorTests testBool]' passed (0.001 seconds).
Test Case '-[YamsTests.ConstructorTests testFloat]' started.
Test Case '-[YamsTests.ConstructorTests testFloat]' passed (0.002 seconds).
Test Case '-[YamsTests.ConstructorTests testInt]' started.
Test Case '-[YamsTests.ConstructorTests testInt]' passed (0.004 seconds).
Test Case '-[YamsTests.ConstructorTests testMap]' started.
Test Case '-[YamsTests.ConstructorTests testMap]' passed (0.001 seconds).
Test Case '-[YamsTests.ConstructorTests testMerge]' started.
Test Case '-[YamsTests.ConstructorTests testMerge]' passed (0.003 seconds).
Test Case '-[YamsTests.ConstructorTests testNull]' started.
Test Case '-[YamsTests.ConstructorTests testNull]' passed (0.001 seconds).
Test Case '-[YamsTests.ConstructorTests testOmap]' started.
Test Case '-[YamsTests.ConstructorTests testOmap]' passed (0.001 seconds).
Test Case '-[YamsTests.ConstructorTests testPairs]' started.
Test Case '-[YamsTests.ConstructorTests testPairs]' passed (0.001 seconds).
Test Case '-[YamsTests.ConstructorTests testQuotationMark]' started.
Test Case '-[YamsTests.ConstructorTests testQuotationMark]' passed (0.001 seconds).
Test Case '-[YamsTests.ConstructorTests testSeq]' started.
Test Case '-[YamsTests.ConstructorTests testSeq]' passed (0.001 seconds).
Test Case '-[YamsTests.ConstructorTests testSet]' started.
Test Case '-[YamsTests.ConstructorTests testSet]' passed (0.002 seconds).
Test Case '-[YamsTests.ConstructorTests testTimestamp]' started.
Test Case '-[YamsTests.ConstructorTests testTimestamp]' passed (0.007 seconds).
Test Case '-[YamsTests.ConstructorTests testTimestampWithNanosecond]' started.
Test Case '-[YamsTests.ConstructorTests testTimestampWithNanosecond]' passed (0.001 seconds).
Test Case '-[YamsTests.ConstructorTests testValue]' started.
Test Case '-[YamsTests.ConstructorTests testValue]' passed (0.003 seconds).
Test Suite 'ConstructorTests' passed at 2018-12-26 19:32:13.136.
	 Executed 15 tests, with 0 failures (0 unexpected) in 0.267 (0.269) seconds
Test Suite 'EmitterTests' started at 2018-12-26 19:32:13.136
Test Case '-[YamsTests.EmitterTests testAllowUnicode]' started.
Test Case '-[YamsTests.EmitterTests testAllowUnicode]' passed (0.001 seconds).
Test Case '-[YamsTests.EmitterTests testLineBreaks]' started.
Test Case '-[YamsTests.EmitterTests testLineBreaks]' passed (0.000 seconds).
Test Case '-[YamsTests.EmitterTests testMapping]' started.
Test Case '-[YamsTests.EmitterTests testMapping]' passed (0.001 seconds).
Test Case '-[YamsTests.EmitterTests testScalar]' started.
Test Case '-[YamsTests.EmitterTests testScalar]' passed (0.000 seconds).
Test Case '-[YamsTests.EmitterTests testSequence]' started.
Test Case '-[YamsTests.EmitterTests testSequence]' passed (0.000 seconds).
Test Case '-[YamsTests.EmitterTests testSortKeys]' started.
Test Case '-[YamsTests.EmitterTests testSortKeys]' passed (0.001 seconds).
Test Suite 'EmitterTests' passed at 2018-12-26 19:32:13.141.
	 Executed 6 tests, with 0 failures (0 unexpected) in 0.005 (0.005) seconds
Test Suite 'EncoderTests' started at 2018-12-26 19:32:13.141
Test Case '-[YamsTests.EncoderTests testDecodingConcreteTypeParameter]' started.
Test Case '-[YamsTests.EncoderTests testDecodingConcreteTypeParameter]' passed (0.004 seconds).
Test Case '-[YamsTests.EncoderTests testDictionary]' started.
Test Case '-[YamsTests.EncoderTests testDictionary]' passed (0.001 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingBase64Data]' started.
Test Case '-[YamsTests.EncoderTests testEncodingBase64Data]' passed (0.001 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingClassWhichSharesEncoderWithSuper]' started.
Test Case '-[YamsTests.EncoderTests testEncodingClassWhichSharesEncoderWithSuper]' passed (0.001 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingDate]' started.
Test Case '-[YamsTests.EncoderTests testEncodingDate]' passed (0.002 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingDateMillisecondsSince1970]' started.
Test Case '-[YamsTests.EncoderTests testEncodingDateMillisecondsSince1970]' passed (0.001 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelDeepStructuredType]' started.
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelDeepStructuredType]' passed (0.003 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelEmptyClass]' started.
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelEmptyClass]' passed (0.001 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelEmptyStruct]' started.
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelEmptyStruct]' passed (0.000 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelNullableType]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:115: error: -[YamsTests.EncoderTests testEncodingTopLevelNullableType] : failed - Failed to decode EnhancedBool from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
true
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:116: error: -[YamsTests.EncoderTests testEncodingTopLevelNullableType] : failed - Failed to decode EnhancedBool from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
false
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:117: error: -[YamsTests.EncoderTests testEncodingTopLevelNullableType] : failed - Failed to decode EnhancedBool from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
null
   ^)))
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelNullableType]' failed (0.009 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelSingleValueClass]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:39: error: -[YamsTests.EncoderTests testEncodingTopLevelSingleValueClass] : failed - Failed to decode Counter from YAML by error: typeMismatch(Swift.Int, Swift.DecodingError.Context(codingPath: [], debugDescription: "Expected to decode Int but found Scalar instead.", underlyingError: nil))
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelSingleValueClass]' failed (0.004 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelSingleValueEnum]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:30: error: -[YamsTests.EncoderTests testEncodingTopLevelSingleValueEnum] : failed - Failed to decode Switch from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
false
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:31: error: -[YamsTests.EncoderTests testEncodingTopLevelSingleValueEnum] : failed - Failed to decode Switch from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
true
   ^)))
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelSingleValueEnum]' failed (0.001 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelSingleValueStruct]' started.
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelSingleValueStruct]' passed (0.003 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelStructuredClass]' started.
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelStructuredClass]' passed (0.001 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelStructuredSingleClass]' started.
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelStructuredSingleClass]' passed (0.003 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelStructuredSingleStruct]' started.
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelStructuredSingleStruct]' passed (0.002 seconds).
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelStructuredStruct]' started.
Test Case '-[YamsTests.EncoderTests testEncodingTopLevelStructuredStruct]' passed (0.001 seconds).
Test Case '-[YamsTests.EncoderTests testInterceptDecimal]' started.
Test Case '-[YamsTests.EncoderTests testInterceptDecimal]' passed (0.002 seconds).
Test Case '-[YamsTests.EncoderTests testInterceptURL]' started.
Test Case '-[YamsTests.EncoderTests testInterceptURL]' passed (0.001 seconds).
Test Case '-[YamsTests.EncoderTests testNestedContainerCodingPaths]' started.
Test Case '-[YamsTests.EncoderTests testNestedContainerCodingPaths]' passed (0.002 seconds).
Test Case '-[YamsTests.EncoderTests testNodeTypeMismatch]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:289: error: -[YamsTests.EncoderTests testNodeTypeMismatch] : failed - unexpected error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
hello
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:289: error: -[YamsTests.EncoderTests testNodeTypeMismatch] : failed - unexpected error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
- hello
   ^)))
Test Case '-[YamsTests.EncoderTests testNodeTypeMismatch]' failed (0.003 seconds).
Test Case '-[YamsTests.EncoderTests testSuperEncoderCodingPaths]' started.
Test Case '-[YamsTests.EncoderTests testSuperEncoderCodingPaths]' passed (0.002 seconds).
Test Case '-[YamsTests.EncoderTests testValuesInKeyedContainer]' started.
Test Case '-[YamsTests.EncoderTests testValuesInKeyedContainer]' passed (0.006 seconds).
Test Case '-[YamsTests.EncoderTests testValuesInSingleValueContainer]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:180: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode Bool from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
true
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:181: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode Bool from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
false
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:184: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode Int8 from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
-128
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:184: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode Int8 from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
127
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:185: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode Int16 from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
-32768
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:185: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode Int16 from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
32767
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:188: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode UInt from YAML by error: valueNotFound(Swift.UInt, Swift.DecodingError.Context(codingPath: [], debugDescription: "Expected UInt value but found Scalar(string: \"읰\", tag: , style: Yams.Node.Scalar.Style.plain, mark: Optional(1:1)) instead.", underlyingError: nil))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:189: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode UInt8 from YAML by error: valueNotFound(Swift.UInt8, Swift.DecodingError.Context(codingPath: [], debugDescription: "Expected UInt8 value but found Scalar(string: \"읰\", tag: , style: Yams.Node.Scalar.Style.plain, mark: Optional(1:1)) instead.", underlyingError: nil))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:189: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode UInt8 from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
255
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:190: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode UInt16 from YAML by error: valueNotFound(Swift.UInt16, Swift.DecodingError.Context(codingPath: [], debugDescription: "Expected UInt16 value but found Scalar(string: \"읰\", tag: , style: Yams.Node.Scalar.Style.plain, mark: Optional(1:1)) instead.", underlyingError: nil))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:190: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode UInt16 from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
65535
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:191: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode UInt32 from YAML by error: valueNotFound(Swift.UInt32, Swift.DecodingError.Context(codingPath: [], debugDescription: "Expected UInt32 value but found Scalar(string: \"읰\", tag: , style: Yams.Node.Scalar.Style.plain, mark: Optional(1:1)) instead.", underlyingError: nil))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:192: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode UInt64 from YAML by error: valueNotFound(Swift.UInt64, Swift.DecodingError.Context(codingPath: [], debugDescription: "Expected UInt64 value but found Scalar(string: \"읰\", tag: , style: Yams.Node.Scalar.Style.plain, mark: Optional(1:1)) instead.", underlyingError: nil))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:194: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode Float from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
.inf
   ^)))
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/EncoderTests.swift:195: error: -[YamsTests.EncoderTests testValuesInSingleValueContainer] : failed - Failed to decode Double from YAML by error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid YAML.", underlyingError: Optional(1:4: error: reader: control characters are not allowed:
.inf
   ^)))
Test Case '-[YamsTests.EncoderTests testValuesInSingleValueContainer]' failed (0.010 seconds).
Test Case '-[YamsTests.EncoderTests testValuesInUnkeyedContainer]' started.
Test Case '-[YamsTests.EncoderTests testValuesInUnkeyedContainer]' passed (0.001 seconds).
Test Suite 'EncoderTests' failed at 2018-12-26 19:32:13.209.
	 Executed 25 tests, with 23 failures (0 unexpected) in 0.066 (0.068) seconds
Test Suite 'MarkTests' started at 2018-12-26 19:32:13.209
Test Case '-[YamsTests.MarkTests testLocatableDeprecationMessageForSwiftLint]' started.
Test Case '-[YamsTests.MarkTests testLocatableDeprecationMessageForSwiftLint]' passed (0.001 seconds).
Test Suite 'MarkTests' passed at 2018-12-26 19:32:13.210.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'NodeTests' started at 2018-12-26 19:32:13.210
Test Case '-[YamsTests.NodeTests testArray]' started.
Test Case '-[YamsTests.NodeTests testArray]' passed (0.001 seconds).
Test Case '-[YamsTests.NodeTests testExpressibleByArrayLiteral]' started.
Test Case '-[YamsTests.NodeTests testExpressibleByArrayLiteral]' passed (0.000 seconds).
Test Case '-[YamsTests.NodeTests testExpressibleByDictionaryLiteral]' started.
Test Case '-[YamsTests.NodeTests testExpressibleByDictionaryLiteral]' passed (0.000 seconds).
Test Case '-[YamsTests.NodeTests testExpressibleByFloatLiteral]' started.
Test Case '-[YamsTests.NodeTests testExpressibleByFloatLiteral]' passed (0.000 seconds).
Test Case '-[YamsTests.NodeTests testExpressibleByIntegerLiteral]' started.
Test Case '-[YamsTests.NodeTests testExpressibleByIntegerLiteral]' passed (0.000 seconds).
Test Case '-[YamsTests.NodeTests testExpressibleByStringLiteral]' started.
Test Case '-[YamsTests.NodeTests testExpressibleByStringLiteral]' passed (0.000 seconds).
Test Case '-[YamsTests.NodeTests testMappingBehavesLikeADictionary]' started.
Test Case '-[YamsTests.NodeTests testMappingBehavesLikeADictionary]' passed (0.001 seconds).
Test Case '-[YamsTests.NodeTests testScalar]' started.
Test Case '-[YamsTests.NodeTests testScalar]' passed (0.000 seconds).
Test Case '-[YamsTests.NodeTests testSequenceBehavesLikeAnArray]' started.
Test Case '-[YamsTests.NodeTests testSequenceBehavesLikeAnArray]' passed (0.001 seconds).
Test Case '-[YamsTests.NodeTests testSubscriptMapping]' started.
Test Case '-[YamsTests.NodeTests testSubscriptMapping]' passed (0.000 seconds).
Test Case '-[YamsTests.NodeTests testSubscriptSequence]' started.
Test Case '-[YamsTests.NodeTests testSubscriptSequence]' passed (0.000 seconds).
Test Case '-[YamsTests.NodeTests testSubscriptWithNonDefaultResolver]' started.
Test Case '-[YamsTests.NodeTests testSubscriptWithNonDefaultResolver]' passed (0.000 seconds).
Test Case '-[YamsTests.NodeTests testTypedAccessorProperties]' started.
Test Case '-[YamsTests.NodeTests testTypedAccessorProperties]' passed (0.001 seconds).
Test Suite 'NodeTests' passed at 2018-12-26 19:32:13.217.
	 Executed 13 tests, with 0 failures (0 unexpected) in 0.006 (0.006) seconds
Test Suite 'PerformanceTests' started at 2018-12-26 19:32:13.217
Test Case '-[YamsTests.PerformanceTests testSourceKittenIssue289UsingCompose]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/PerformanceTests.swift:92: Test Case '-[YamsTests.PerformanceTests testSourceKittenIssue289UsingCompose]' measured [Time, seconds] average: 0.012, relative standard deviation: 17.768%, values: [0.014658, 0.010010, 0.011343, 0.014352, 0.009674, 0.009873, 0.015215, 0.013424, 0.010828, 0.009920], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100
Test Case '-[YamsTests.PerformanceTests testSourceKittenIssue289UsingCompose]' passed (0.406 seconds).
Test Case '-[YamsTests.PerformanceTests testSourceKittenIssue289UsingLoad]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/PerformanceTests.swift:66: Test Case '-[YamsTests.PerformanceTests testSourceKittenIssue289UsingLoad]' measured [Time, seconds] average: 0.016, relative standard deviation: 7.909%, values: [0.019358, 0.015514, 0.015583, 0.014824, 0.015543, 0.015097, 0.015140, 0.016230, 0.015351, 0.015128], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100
Test Case '-[YamsTests.PerformanceTests testSourceKittenIssue289UsingLoad]' passed (0.413 seconds).
Test Case '-[YamsTests.PerformanceTests testSourceKittenIssue289UsingSwiftDecodable]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/PerformanceTests.swift:118: Test Case '-[YamsTests.PerformanceTests testSourceKittenIssue289UsingSwiftDecodable]' measured [Time, seconds] average: 0.013, relative standard deviation: 17.285%, values: [0.015417, 0.013634, 0.011615, 0.013201, 0.014060, 0.011517, 0.011462, 0.011756, 0.019267, 0.012169], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100
Test Case '-[YamsTests.PerformanceTests testSourceKittenIssue289UsingSwiftDecodable]' passed (0.388 seconds).
Test Suite 'PerformanceTests' passed at 2018-12-26 19:32:14.424.
	 Executed 3 tests, with 0 failures (0 unexpected) in 1.207 (1.208) seconds
Test Suite 'RepresenterTests' started at 2018-12-26 19:32:14.425
Test Case '-[YamsTests.RepresenterTests testArray]' started.
Test Case '-[YamsTests.RepresenterTests testArray]' passed (0.000 seconds).
Test Case '-[YamsTests.RepresenterTests testBool]' started.
Test Case '-[YamsTests.RepresenterTests testBool]' passed (0.000 seconds).
Test Case '-[YamsTests.RepresenterTests testData]' started.
Test Case '-[YamsTests.RepresenterTests testData]' passed (0.000 seconds).
Test Case '-[YamsTests.RepresenterTests testDate]' started.
Test Case '-[YamsTests.RepresenterTests testDate]' passed (0.001 seconds).
Test Case '-[YamsTests.RepresenterTests testDictionary]' started.
Test Case '-[YamsTests.RepresenterTests testDictionary]' passed (0.001 seconds).
Test Case '-[YamsTests.RepresenterTests testDouble]' started.
Test Case '-[YamsTests.RepresenterTests testDouble]' passed (0.000 seconds).
Test Case '-[YamsTests.RepresenterTests testFloat]' started.
Test Case '-[YamsTests.RepresenterTests testFloat]' passed (0.000 seconds).
Test Case '-[YamsTests.RepresenterTests testInteger]' started.
Test Case '-[YamsTests.RepresenterTests testInteger]' passed (0.001 seconds).
Test Case '-[YamsTests.RepresenterTests testOptional]' started.
Test Case '-[YamsTests.RepresenterTests testOptional]' passed (0.000 seconds).
Test Case '-[YamsTests.RepresenterTests testString]' started.
Test Case '-[YamsTests.RepresenterTests testString]' passed (0.000 seconds).
Test Suite 'RepresenterTests' passed at 2018-12-26 19:32:14.429.
	 Executed 10 tests, with 0 failures (0 unexpected) in 0.004 (0.004) seconds
Test Suite 'ResolverTests' started at 2018-12-26 19:32:14.429
Test Case '-[YamsTests.ResolverTests testBasic]' started.
Test Case '-[YamsTests.ResolverTests testBasic]' passed (0.000 seconds).
Test Case '-[YamsTests.ResolverTests testCustomize]' started.
<unknown>:0: error: -[YamsTests.ResolverTests testCustomize] : failed: caught error: The operation couldn’t be completed. (Yams.YamlError error 0.)
Test Case '-[YamsTests.ResolverTests testCustomize]' failed (0.006 seconds).
Test Case '-[YamsTests.ResolverTests testDefault]' started.
Test Case '-[YamsTests.ResolverTests testDefault]' passed (0.001 seconds).
Test Suite 'ResolverTests' failed at 2018-12-26 19:32:14.436.
	 Executed 3 tests, with 1 failure (1 unexpected) in 0.007 (0.007) seconds
Test Suite 'SpecTests' started at 2018-12-26 19:32:14.436
Test Case '-[YamsTests.SpecTests testEmptyString]' started.
Test Case '-[YamsTests.SpecTests testEmptyString]' passed (0.000 seconds).
Test Case '-[YamsTests.SpecTests testMultibyteCharacters]' started.
Test Case '-[YamsTests.SpecTests testMultibyteCharacters]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_10_NodeForSammySosaAppearsTwiceInThisDocument]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_10_NodeForSammySosaAppearsTwiceInThisDocument]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_11_MappingBetweenSequences]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_11_MappingBetweenSequences]' passed (0.000 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_12_CompactNestedMapping]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_12_CompactNestedMapping]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_13_Inliterals_NewlinesArePreserved]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_13_Inliterals_NewlinesArePreserved]' passed (0.000 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_14_InTheFoldedScalars_NewlinesBecomeSpaces]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_14_InTheFoldedScalars_NewlinesBecomeSpaces]' passed (0.000 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_15_InTheFoldedScalars_NewlinesBecomeSpaces]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_15_InTheFoldedScalars_NewlinesBecomeSpaces]' passed (0.000 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_16_IndentationDeterminesScope]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_16_IndentationDeterminesScope]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_17_QuotedScalars]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_17_QuotedScalars]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_18_MultiLineFlowScalars]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_18_MultiLineFlowScalars]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_19_Integers]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_19_Integers]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_1_SequenceOfScalars]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_1_SequenceOfScalars]' passed (0.000 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_20_FloatingPoint]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_20_FloatingPoint]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_23_VariousExplicitTags]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_23_VariousExplicitTags]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_27_Invoice]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_27_Invoice]' passed (0.006 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_28_LogFile]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_28_LogFile]' passed (0.007 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_2_MappingScalarsToScalars]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_2_MappingScalarsToScalars]' passed (0.002 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_3_MappingScalarsToSequences]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_3_MappingScalarsToSequences]' passed (0.002 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_4_SequenceOfMappings]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_4_SequenceOfMappings]' passed (0.002 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_5_SequenceOfSequences]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_5_SequenceOfSequences]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_6_MappingOfMappings]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_6_MappingOfMappings]' passed (0.002 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_7_TwoDocumentsInAStream]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_7_TwoDocumentsInAStream]' passed (0.001 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_8_PlayByPlayFeedFromAGame]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_8_PlayByPlayFeedFromAGame]' passed (0.002 seconds).
Test Case '-[YamsTests.SpecTests testSpecExample2_9_SingleDocumentWithTwoComments]' started.
Test Case '-[YamsTests.SpecTests testSpecExample2_9_SingleDocumentWithTwoComments]' passed (0.001 seconds).
Test Suite 'SpecTests' passed at 2018-12-26 19:32:14.472.
	 Executed 25 tests, with 0 failures (0 unexpected) in 0.034 (0.036) seconds
Test Suite 'StringTests' started at 2018-12-26 19:32:14.472
Test Case '-[YamsTests.StringTests testConfirmBehaviorOfStandardLibraryAPI]' started.
Test Case '-[YamsTests.StringTests testConfirmBehaviorOfStandardLibraryAPI]' passed (0.000 seconds).
Test Case '-[YamsTests.StringTests testSubstringAtLine]' started.
Test Case '-[YamsTests.StringTests testSubstringAtLine]' passed (0.000 seconds).
Test Case '-[YamsTests.StringTests testUTF16LineNumberColumnAndContentsAtOffset]' started.
Test Case '-[YamsTests.StringTests testUTF16LineNumberColumnAndContentsAtOffset]' passed (0.000 seconds).
Test Case '-[YamsTests.StringTests testUTF8LineNumberColumnAndContentsAtOffset]' started.
Test Case '-[YamsTests.StringTests testUTF8LineNumberColumnAndContentsAtOffset]' passed (0.000 seconds).
Test Suite 'StringTests' passed at 2018-12-26 19:32:14.473.
	 Executed 4 tests, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'YamlErrorTests' started at 2018-12-26 19:32:14.474
Test Case '-[YamsTests.YamlErrorTests testNextRootThrowsOnInvalidYaml]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/YamlErrorTests.swift:74: error: -[YamsTests.YamlErrorTests testNextRootThrowsOnInvalidYaml] : XCTAssertEqual failed: ("Optional(Yams.Node.scalar(Yams.Node.Scalar(string: "흀翾", tag: , style: Yams.Node.Scalar.Style.plain, mark: Optional(1:1))))") is not equal to ("Optional(Yams.Node.scalar(Yams.Node.Scalar(string: "", tag: tag:yaml.org,2002:str, style: Yams.Node.Scalar.Style.literal, mark: nil)))") - 
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/YamlErrorTests.swift:76: error: -[YamsTests.YamlErrorTests testNextRootThrowsOnInvalidYaml] : XCTAssertThrowsError failed: did not throw an error - 
Test Case '-[YamsTests.YamlErrorTests testNextRootThrowsOnInvalidYaml]' failed (0.001 seconds).
Test Case '-[YamsTests.YamlErrorTests testScannerErrorMayHaveNullContext]' started.
Test Case '-[YamsTests.YamlErrorTests testScannerErrorMayHaveNullContext]' passed (0.000 seconds).
Test Case '-[YamsTests.YamlErrorTests testSingleRootThrowsOnInvalidYaml]' started.
/Users/norio/github/swift-dev/SourceKitten/Carthage/Checkouts/Yams/Tests/YamsTests/YamlErrorTests.swift:91: error: -[YamsTests.YamlErrorTests testSingleRootThrowsOnInvalidYaml] : XCTAssertThrowsError failed: did not throw an error - 
Test Case '-[YamsTests.YamlErrorTests testSingleRootThrowsOnInvalidYaml]' failed (0.000 seconds).
Test Case '-[YamsTests.YamlErrorTests testSingleRootThrowsOnMultipleDocuments]' started.
Test Case '-[YamsTests.YamlErrorTests testSingleRootThrowsOnMultipleDocuments]' passed (0.000 seconds).
Test Case '-[YamsTests.YamlErrorTests testUndefinedAliasCausesError]' started.
Test Case '-[YamsTests.YamlErrorTests testUndefinedAliasCausesError]' passed (0.000 seconds).
Test Case '-[YamsTests.YamlErrorTests testYamlErrorEmitter]' started.
Test Case '-[YamsTests.YamlErrorTests testYamlErrorEmitter]' passed (0.000 seconds).
Test Case '-[YamsTests.YamlErrorTests testYamlErrorParser]' started.
Test Case '-[YamsTests.YamlErrorTests testYamlErrorParser]' passed (0.000 seconds).
Test Case '-[YamsTests.YamlErrorTests testYamlErrorReader]' started.
Test Case '-[YamsTests.YamlErrorTests testYamlErrorReader]' passed (0.000 seconds).
Test Case '-[YamsTests.YamlErrorTests testYamlErrorScanner]' started.
Test Case '-[YamsTests.YamlErrorTests testYamlErrorScanner]' passed (0.000 seconds).
Test Suite 'YamlErrorTests' failed at 2018-12-26 19:32:14.478.
	 Executed 9 tests, with 3 failures (0 unexpected) in 0.004 (0.004) seconds
Test Suite 'YamsPackageTests.xctest' failed at 2018-12-26 19:32:14.478.
	 Executed 114 tests, with 27 failures (1 unexpected) in 1.600 (1.611) seconds
Test Suite 'All tests' failed at 2018-12-26 19:32:14.478.
	 Executed 114 tests, with 27 failures (1 unexpected) in 1.600 (1.611) seconds
xcrun --toolchain org.swift.5020181225a swift test  20.24s user 2.76s system 193% cpu 11.877 total
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,7 +10,7 @@
 
 ##### Bug Fixes
 
-* Fix some tests fail with Swift 5.0 on macOS.  
+* Fix some test failures with the latest Swift 5 snapshot on Apple platforms.  
   [Norio Nomura](https://github.com/norio-nomura)
   [#143](https://github.com/jpsim/Yams/issues/143)
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,7 +10,9 @@
 
 ##### Bug Fixes
 
-* None.
+* Fix some tests fail with Swift 5.0 on macOS.  
+  [Norio Nomura](https://github.com/norio-nomura)
+  [#143](https://github.com/jpsim/Yams/issues/143)
 
 ## 1.0.1
 

--- a/Sources/Yams/Parser.swift
+++ b/Sources/Yams/Parser.swift
@@ -125,22 +125,15 @@ public final class Parser {
         yaml_parser_initialize(&parser)
 #if USE_UTF8
         yaml_parser_set_encoding(&parser, YAML_UTF8_ENCODING)
-        utf8CString = string.utf8CString
-        try utf8CString.withUnsafeBytes { (bytes: UnsafeRawBufferPointer) in
-            let input = bytes.baseAddress?.assumingMemoryBound(to: UInt8.self)
-            yaml_parser_set_input_string(&parser, input, bytes.count - 1)
-            try parse() // Drop YAML_STREAM_START_EVENT
-        }
+        utf8Slice = string.utf8CString.dropLast()
+        try utf8Slice.withUnsafeBytes(startParse(with:))
 #else
         // use native endian
         let isLittleEndian = 1 == 1.littleEndian
         yaml_parser_set_encoding(&parser, isLittleEndian ? YAML_UTF16LE_ENCODING : YAML_UTF16BE_ENCODING)
         let encoding: String.Encoding = isLittleEndian ? .utf16LittleEndian : .utf16BigEndian
         data = yaml.data(using: encoding)!
-        try data.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) in
-            yaml_parser_set_input_string(&parser, bytes, data.count)
-            try parse() // Drop YAML_STREAM_START_EVENT
-        }
+        try data.withUnsafeBytes(startParse(with:))
 #endif
     }
 
@@ -183,7 +176,7 @@ public final class Parser {
     private var anchors = [String: Node]()
     private var parser = yaml_parser_t()
 #if USE_UTF8
-    private let utf8CString: ContiguousArray<CChar>
+    private let utf8Slice: ArraySlice<CChar>
 #else
     private let data: Data
 #endif
@@ -214,6 +207,11 @@ private extension Parser {
         default:
             fatalError("unreachable")
         }
+    }
+
+    func startParse(with buffer: UnsafeRawBufferPointer) throws {
+        yaml_parser_set_input_string(&parser, buffer.baseAddress?.assumingMemoryBound(to: UInt8.self), buffer.count)
+        try parse() // Drop YAML_STREAM_START_EVENT
     }
 
     @discardableResult
@@ -353,3 +351,24 @@ private class Event {
 private func string(from pointer: UnsafePointer<UInt8>!) -> String? {
     return String.decodeCString(pointer, as: UTF8.self, repairingInvalidCodeUnits: true)?.result
 }
+
+#if swift(>=4.2)
+#if compiler(>=5) && canImport(Darwin)
+#else
+private extension Data {
+    func withUnsafeBytes<Result>(_ apply: (UnsafeRawBufferPointer) throws -> Result) rethrows -> Result {
+        return try withUnsafeBytes {
+            try apply(UnsafeRawBufferPointer(start: $0, count: count))
+        }
+    }
+}
+#endif
+#else
+private extension Data {
+    func withUnsafeBytes<Result>(_ apply: (UnsafeRawBufferPointer) throws -> Result) rethrows -> Result {
+        return try withUnsafeBytes {
+            try apply(UnsafeRawBufferPointer(start: $0, count: count))
+        }
+    }
+}
+#endif

--- a/Sources/Yams/Parser.swift
+++ b/Sources/Yams/Parser.swift
@@ -126,9 +126,10 @@ public final class Parser {
 #if USE_UTF8
         yaml_parser_set_encoding(&parser, YAML_UTF8_ENCODING)
         utf8CString = string.utf8CString
-        utf8CString.withUnsafeBytes { bytes in
+        try utf8CString.withUnsafeBytes { (bytes: UnsafeRawBufferPointer) in
             let input = bytes.baseAddress?.assumingMemoryBound(to: UInt8.self)
             yaml_parser_set_input_string(&parser, input, bytes.count - 1)
+            try parse() // Drop YAML_STREAM_START_EVENT
         }
 #else
         // use native endian
@@ -136,11 +137,11 @@ public final class Parser {
         yaml_parser_set_encoding(&parser, isLittleEndian ? YAML_UTF16LE_ENCODING : YAML_UTF16BE_ENCODING)
         let encoding: String.Encoding = isLittleEndian ? .utf16LittleEndian : .utf16BigEndian
         data = yaml.data(using: encoding)!
-        data.withUnsafeBytes { bytes in
+        try data.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) in
             yaml_parser_set_input_string(&parser, bytes, data.count)
+            try parse() // Drop YAML_STREAM_START_EVENT
         }
 #endif
-        try parse() // Drop YAML_STREAM_START_EVENT
     }
 
     deinit {

--- a/Sources/Yams/Parser.swift
+++ b/Sources/Yams/Parser.swift
@@ -353,8 +353,7 @@ private func string(from pointer: UnsafePointer<UInt8>!) -> String? {
 }
 
 #if swift(>=4.2)
-#if compiler(>=5) && canImport(Darwin)
-#else
+#if compiler(<5) || !canImport(Darwin)
 private extension Data {
     func withUnsafeBytes<Result>(_ apply: (UnsafeRawBufferPointer) throws -> Result) rethrows -> Result {
         return try withUnsafeBytes {

--- a/Sources/Yams/Parser.swift
+++ b/Sources/Yams/Parser.swift
@@ -353,7 +353,7 @@ private func string(from pointer: UnsafePointer<UInt8>!) -> String? {
 }
 
 #if swift(>=4.2)
-#if compiler(<5) || !canImport(Darwin)
+#if !compiler(>=5) || !canImport(Darwin)
 private extension Data {
     func withUnsafeBytes<Result>(_ apply: (UnsafeRawBufferPointer) throws -> Result) rethrows -> Result {
         return try withUnsafeBytes {

